server/v2/server.go

```diff
@@ -111,14 +121,20 @@ func (s *Server[T]) Stop(ctx context.Context) error {
 	logger := GetLoggerFromContext(ctx)
 	logger.With(log.ModuleKey, s.Name()).Info("stopping servers...")
 
-	g, ctx := errgroup.WithContext(ctx)
+	// cannot reuse context as it's already canceled, so create a new one for stopping
```

Comment:
As this resolves https://github.com/cosmos/cosmos-sdk/issues/22405, could you extract in a seperate PR? It is unrelated to the genesis issue

Reply:
There's still a problem with it... `Stop()` can be called before `Start()` completes. >.<
I think the start command needs another look. I will probably reverse this change in this branch it's helpful during development.

Reply:
Isn't this super easily fixable by deferring the Stop instead of doing what we do now?